contracts/mainnet/connectors/morpho/main.sol

```diff
@@ -0,0 +1,176 @@
+
+        setUint(_setId, _amt);
+
+        _eventName = "LogBorrow(uint256,address,uint256,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_pool,
+			_poolTokenAddress,
+			_amt,
+			_maxGasForMatching,
+			_getId,
+			_setId
+		);
+    }
+
+    function withdraw (
+        Underlying _pool,
+        address _tokenAddress,
+        address _poolTokenAddress, // if address weth (send eth)
+        uint256 _amount, // amount max
+        uint256 _getId,
+        uint256 _setId
+    ) external payable returns(string memory _eventName, bytes memory _eventParam) {
+
+        require(_pool == Underlying.AAVEV2 || _pool == Underlying.COMPOUNDV2, 'protocol not supported');
+
+        uint256 _amt = getUint(_getId, _amount);
+        bool _isETH = _tokenAddress == ethAddr;
+        address _token = _isETH ? wethAddr : _tokenAddress;
+
+        if (_amt == uint256(-1)) _amt = _poolTokenAddress.balanceOf(address(this));
+
+        if(_pool == Underlying.AAVEV2) {
+            morphoAave.withdraw(_poolTokenAddress, _amt);
+        } else {
+            morphoCompound.withdraw(_poolTokenAddress, _amt);
+        }
+
+        convertWethToEth(_isETH, TokenInterface(_token), _amt);
+
+        setUint(_setId, _amt);
+
+		_eventName = "LogWithdraw(uint256,bool,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(_pool, _isETH, _poolTokenAddress, _amt, _getId, _setId);
+
+    }
+
+    function payback (
+        Underlying _pool,
+        address _tokenAddress,
+        address _poolTokenAddress, // if address weth (send eth)
+        uint256 _amount, // max value
+        uint256 _getId,
+        uint256 _setId
+    ) external payable returns(string memory _eventName, bytes memory _eventParam) {
+
+        require(_pool == Underlying.AAVEV2 || _pool == Underlying.COMPOUNDV2, 'protocol not supported');
+
+        bool _isETH = _tokenAddress == ethAddr;
+        uint256 _amt = getUint(_getId, _amount);
+        address _token = _isETH ? wethAddr : _tokenAddress;
+
+        if(_amt == uint256(-1)) {
+            _amt = _isETH ? _amt = address(this).balance : TokenInterface(_token).balanceOf(address(this));
+        }
+
+        if (_isETH) convertEthToWeth(_isETH, TokenInterface(_token), _amt);
+
+        _pool == Underlying.AAVEV2  
+        ? morphoAave.repay(_poolTokenAddress, address(this), _amt)
+        : morphoCompound.repay(_poolTokenAddress, address(this), _amt);
+
+        setUint(_setId, _amt);
+
+		_eventName = "LogPayback(uint256,bool,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(_pool, _isETH, _poolTokenAddress, _amt, _getId, _setId);
+    }
+
+    function claim (
+        Underlying _pool,
+        address[] _tokenAddresses, //todo: eth will be claimed as weth currently?
+        bool _tradeForMorphoToken
+    ) external payable returns(string memory _eventName, bytes memory _eventParam) {
+
+        require(_pool == Underlying.AAVEV2 || _pool == Underlying.COMPOUNDV2, 'protocol not supported');
```

Comment:
@shriyatyagii is there a way to check if a pool is of Morpho or not? Instead of hardcoding?

Reply:
@thrilok209 Morpho has different contracts for different protocols. Currently couldn't find any function to give on-chain data whether a protocol is supported by Morpho.
Will ask the team once.